internal/driver/fetch.go

```diff
@@ -41,36 +41,119 @@ import (
 // there are some failures. It will return an error if it is unable to
 // fetch any profiles.
 func fetchProfiles(s *source, o *plugin.Options) (*profile.Profile, error) {
-	sources := make([]profileSource, 0, len(s.Sources)+len(s.Base))
-	for _, src := range s.Sources {
-		sources = append(sources, profileSource{
-			addr:   src,
-			source: s,
-			scale:  1,
-		})
-	}
-	for _, src := range s.Base {
-		sources = append(sources, profileSource{
-			addr:   src,
-			source: s,
-			scale:  -1,
-		})
-	}
-	p, msrcs, save, cnt, err := chunkedGrab(sources, o.Fetch, o.Obj, o.UI)
-	if err != nil {
-		return nil, err
-	}
-	if cnt == 0 {
-		return nil, fmt.Errorf("failed to fetch any profiles")
-	}
-	if want, got := len(sources), cnt; want != got {
-		o.UI.PrintErr(fmt.Sprintf("fetched %d profiles out of %d", got, want))
+	normalize := s.Normalize
+
+	var p *profile.Profile
+	var save bool
+	var msrcs plugin.MappingSources
+
+	if normalize {
```

Comment:
Can you limit this condition to only whether the profile is normalized or not? There is no reason for the fetch of the profiles to be different that much, is there?